src/lib/isMagnetURI.js

```diff
@@ -1,6 +1,6 @@
 import assertString from './util/assertString';
 
-const magnetURI = /^magnet:\?xt=urn:[a-z0-9]+:[a-z0-9]{32,40}&dn=.+&tr=.+$/i;
+const magnetURI = /^magnet:\?xt(\.1)?=urn:(?:aich|bitprint|btih|ed2k|ed2khash|kzhash|md5|sha1|tree:tiger):[a-z0-9]{32}(?:[a-z0-9]{8})?(?:&.+)?/i;
```

Comment:
One nit: `(?:&.+)`, should we put a range limit for `+`?

Reply:
Fixed, we only need to check that the string is ending or there is a parameter start ( `magnet:?xt=urn:ed2k:ABCDEFGHIJKLMNOPQRSTUVWXYZ12345678901234&` is a valid magnet uri )